deskthing: init at 0.11.17

malmeloo opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Adds [deskthing](https://deskthing.app/). Supersedes #372690.

This desktop app can (for the most part) be tested without owning a physical Car Thing:
- Launch the program
  - It will show file not found errors in the terminal, this seems to be an issue with the program itself (restarting fixes it)
- Download the Deskthing Client by going to `Downloads -> Client` (the `Client` button is a dropdown under the `Downloads` button in the header)
  - You can also follow the "setting up a client" task if you click on the `Tasks` button in the bottom left
- Restart the server (`Clients` menu then `Restart Server` in the bottom left) or the entire application (make sure to quit it entirely, simply closing the window will just put it in the background)
- Visit `http://localhost:8891/` in a web browser
- Install some apps from the `Downloads` page if you wish

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

malmeloo committed:
maintainers: add malmeloo

Sigmanificient commented:
> This desktop app can (for the most part) be tested without owning a physical Car Thing: simply visit http://localhost:8891/ in a web browser after having launched the program.

![image](https://github.com/user-attachments/assets/0eb4c7d4-f644-4abf-8df1-3ad70054207f)

# 

I also have the following log that may indicate a missing dependency?

```
server 5/22/2025, 12:42:13 AM ERROR [unknown]: No audio source found. Please install an audio app.
```

malmeloo commented:
Ah sorry, it takes a little more effort to set up from a fresh install. I've updated the PR body with better instructions 🙂

They also released some new versions recently, so I'll push an update in a second.

malmeloo commented:
Updated to v0.11.10 and fixed the desktop icon which was apparently broken.

If you had a lower version installed before you might want to wipe your config: `rm -rf ~/.config/deskthing`. Deskthing does not appear to handle config migrations well.

Also new in this version: if no apps are shown under the "Downloads" menu or you get other weird errors, you might need to use the "Refresh" button on the left a few times. It's... not the most stable program.

Iamanaws commented on pkgs/by-name/de/deskthing/package.nix:
```diff
@@ -0,0 +1,48 @@
+
+  meta = {
+    homepage = "https://deskthing.app/";
+    downloadPage = "https://github.com/ItsRiprod/${pname}/releases/";
```
```suggestion
    downloadPage = "https://github.com/ItsRiprod/deskthing/releases/";
```

iamanaws reviewed (commented):
Some changes:
- avoid using pname on meta
- please squash/fixup the `deskthing: init` commits, so it only remains the maintainer commit + a simple pkg init commit

malmeloo committed:
deskthing: init at 0.11.17

malmeloo commented:
Thanks for the feedback! I've applied your suggestions and bumped to 0.11.14 in the process.

iamanaws reviewed (approved):
## `nixpkgs-review` result for [#406532](https://github.com/NixOS/nixpkgs/pull/406532)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 406532`

Logs: https://github.com/Iamanaws/nixpkgs-review-gha/actions/runs/16708539968


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>deskthing</li>
  </ul>
</details>
